File: app-android/app/src/main/java/com/health2u/domain/repository/HealthInsightRepository.kt
package com.health2u.domain.repository

import com.health2u.data.Result
import com.health2u.domain.model.HealthInsight
import kotlinx.coroutines.flow.Flow

interface HealthInsightRepository {
    suspend fun getHealthInsights(): Result<List<HealthInsight>>
    suspend fun getHealthInsightById(id: String): Result<HealthInsight>
    fun observeHealthInsights(): Flow<List<HealthInsight>>
}

File: app-ios/Sources/Health2u/Domain/Repository/HealthInsightRepository.swift
import Foundation

public protocol HealthInsightRepository: Sendable {
    func getHealthInsights() async -> Result<[HealthInsight], APIError>
    func getHealthInsight(id: String) async -> Result<HealthInsight, APIError>
    func observeHealthInsights() -> AsyncStream<[HealthInsight]>
}
